NPEs when initializing items

The demo setup of openHAB now results in such exceptions in the log:

```
2017-05-25 17:28:58.990 [ERROR] [ore.internal.events.OSGiEventManager] - Dispatching/filtering event for subscriber 'org.eclipse.smarthome.core.events.EventSubscriber' failed: java.lang.IllegalArgumentException: The argument 'state' must not be null or empty.
java.util.concurrent.ExecutionException: java.lang.IllegalArgumentException: The argument 'state' must not be null or empty.
        at java.util.concurrent.FutureTask.report(FutureTask.java:122)[:1.8.0_112]
        at java.util.concurrent.FutureTask.get(FutureTask.java:206)[:1.8.0_112]
        at org.eclipse.smarthome.core.common.SafeMethodCaller.callAsynchronous(SafeMethodCaller.java:194)[99:org.eclipse.smarthome.core:0.9.0.201705251309]
        at org.eclipse.smarthome.core.common.SafeMethodCaller.call(SafeMethodCaller.java:83)[99:org.eclipse.smarthome.core:0.9.0.201705251309]
        at org.eclipse.smarthome.core.common.SafeMethodCaller.call(SafeMethodCaller.java:67)[99:org.eclipse.smarthome.core:0.9.0.201705251309]
        at org.eclipse.smarthome.core.internal.events.OSGiEventManager.dispatchESHEvent(OSGiEventManager.java:188)[99:org.eclipse.smarthome.core:0.9.0.201705251309]
        at org.eclipse.smarthome.core.internal.events.OSGiEventManager.handleEvent(OSGiEventManager.java:163)[99:org.eclipse.smarthome.core:0.9.0.201705251309]
        at org.eclipse.smarthome.core.internal.events.OSGiEventManager.handleEvent(OSGiEventManager.java:146)[99:org.eclipse.smarthome.core:0.9.0.201705251309]
        at org.apache.felix.eventadmin.impl.handler.EventHandlerProxy.sendEvent(EventHandlerProxy.java:415)[6:org.apache.karaf.services.eventadmin:4.0.8]
        at org.apache.felix.eventadmin.impl.tasks.HandlerTask.run(HandlerTask.java:90)[6:org.apache.karaf.services.eventadmin:4.0.8]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)[:1.8.0_112]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)[:1.8.0_112]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)[:1.8.0_112]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)[:1.8.0_112]
        at java.lang.Thread.run(Thread.java:745)[:1.8.0_112]
Caused by: java.lang.IllegalArgumentException: The argument 'state' must not be null or empty.
        at com.google.common.base.Preconditions.checkArgument(Preconditions.java:122)[12:com.google.guava:18.0.0]
        at org.eclipse.smarthome.core.items.events.ItemEventFactory.assertValidArguments(ItemEventFactory.java:370)[99:org.eclipse.smarthome.core:0.9.0.201705251309]
        at org.eclipse.smarthome.core.items.events.ItemEventFactory.createStateChangedEvent(ItemEventFactory.java:272)[99:org.eclipse.smarthome.core:0.9.0.201705251309]
        at org.eclipse.smarthome.core.items.GenericItem.sendStateChangedEvent(GenericItem.java:215)[99:org.eclipse.smarthome.core:0.9.0.201705251309]
        at org.eclipse.smarthome.core.items.GenericItem.applyState(GenericItem.java:209)[99:org.eclipse.smarthome.core:0.9.0.201705251309]
        at org.eclipse.smarthome.core.library.items.DimmerItem.setState(DimmerItem.java:74)[99:org.eclipse.smarthome.core:0.9.0.201705251309]
        at org.eclipse.smarthome.core.internal.items.ItemUpdater.receiveUpdate(ItemUpdater.java:70)[99:org.eclipse.smarthome.core:0.9.0.201705251309]
        at org.eclipse.smarthome.core.items.events.AbstractItemEventSubscriber.receive(AbstractItemEventSubscriber.java:44)[99:org.eclipse.smarthome.core:0.9.0.201705251309]
        at org.eclipse.smarthome.core.internal.events.OSGiEventManager$1.call(OSGiEventManager.java:192)[99:org.eclipse.smarthome.core:0.9.0.201705251309]
        at org.eclipse.smarthome.core.internal.events.OSGiEventManager$1.call(OSGiEventManager.java:1)[99:org.eclipse.smarthome.core:0.9.0.201705251309]
        at org.eclipse.smarthome.core.common.SafeMethodCaller$CallableWrapper.call(SafeMethodCaller.java:181)[99:org.eclipse.smarthome.core:0.9.0.201705251309]
        ... 4 more
2017-05-25 17:28:59.064 [ERROR] [ore.internal.events.OSGiEventManager] - Dispatching/filtering event for subscriber 'org.eclipse.smarthome.core.events.EventSubscriber' failed: java.lang.NullPointerException
java.util.concurrent.ExecutionException: java.lang.NullPointerException
        at java.util.concurrent.FutureTask.report(FutureTask.java:122)[:1.8.0_112]
        at java.util.concurrent.FutureTask.get(FutureTask.java:206)[:1.8.0_112]
        at org.eclipse.smarthome.core.common.SafeMethodCaller.callAsynchronous(SafeMethodCaller.java:194)[99:org.eclipse.smarthome.core:0.9.0.201705251309]
        at org.eclipse.smarthome.core.common.SafeMethodCaller.call(SafeMethodCaller.java:83)[99:org.eclipse.smarthome.core:0.9.0.201705251309]
        at org.eclipse.smarthome.core.common.SafeMethodCaller.call(SafeMethodCaller.java:67)[99:org.eclipse.smarthome.core:0.9.0.201705251309]
        at org.eclipse.smarthome.core.internal.events.OSGiEventManager.dispatchESHEvent(OSGiEventManager.java:188)[99:org.eclipse.smarthome.core:0.9.0.201705251309]
        at org.eclipse.smarthome.core.internal.events.OSGiEventManager.handleEvent(OSGiEventManager.java:163)[99:org.eclipse.smarthome.core:0.9.0.201705251309]
        at org.eclipse.smarthome.core.internal.events.OSGiEventManager.handleEvent(OSGiEventManager.java:146)[99:org.eclipse.smarthome.core:0.9.0.201705251309]
        at org.apache.felix.eventadmin.impl.handler.EventHandlerProxy.sendEvent(EventHandlerProxy.java:415)[6:org.apache.karaf.services.eventadmin:4.0.8]
        at org.apache.felix.eventadmin.impl.tasks.HandlerTask.run(HandlerTask.java:90)[6:org.apache.karaf.services.eventadmin:4.0.8]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)[:1.8.0_112]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)[:1.8.0_112]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)[:1.8.0_112]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)[:1.8.0_112]
        at java.lang.Thread.run(Thread.java:745)[:1.8.0_112]
Caused by: java.lang.NullPointerException
        at org.eclipse.smarthome.core.items.GenericItem.applyState(GenericItem.java:208)[99:org.eclipse.smarthome.core:0.9.0.201705251309]
        at org.eclipse.smarthome.core.library.items.DimmerItem.setState(DimmerItem.java:74)[99:org.eclipse.smarthome.core:0.9.0.201705251309]
        at org.eclipse.smarthome.core.internal.items.ItemUpdater.receiveUpdate(ItemUpdater.java:70)[99:org.eclipse.smarthome.core:0.9.0.201705251309]
        at org.eclipse.smarthome.core.items.events.AbstractItemEventSubscriber.receive(AbstractItemEventSubscriber.java:44)[99:org.eclipse.smarthome.core:0.9.0.201705251309]
        at org.eclipse.smarthome.core.internal.events.OSGiEventManager$1.call(OSGiEventManager.java:192)[99:org.eclipse.smarthome.core:0.9.0.201705251309]
        at org.eclipse.smarthome.core.internal.events.OSGiEventManager$1.call(OSGiEventManager.java:1)[99:org.eclipse.smarthome.core:0.9.0.201705251309]
        at org.eclipse.smarthome.core.common.SafeMethodCaller$CallableWrapper.call(SafeMethodCaller.java:181)[99:org.eclipse.smarthome.core:0.9.0.201705251309]
        ... 4 more
```

See also https://community.openhab.org/t/java-lang-nullpointerexception-every-few-seconds/28944

--- a/bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/ItemUtil.java
+++ b/bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/ItemUtil.java
@@ -7,7 +7,10 @@
  */
 package org.eclipse.smarthome.core.items;
 
+import org.apache.commons.lang.StringUtils;
 import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
@@ -17,8 +20,10 @@ import org.slf4j.LoggerFactory;
  *
  * @author Michael Grammling - Initial contribution and API
  * @author Simon Kaufmann - added type conversion
+ * @author Martin van Wingerden - when converting types convert null to UnDefType.NULL
  */
 public class ItemUtil {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ItemUtil.class);
 
     /**
      * The constructor is private.
@@ -44,11 +49,7 @@ public class ItemUtil {
      * @return true if the specified name is a valid item name, otherwise false
      */
     public static boolean isValidItemName(final String itemName) {
-        if (itemName != null && !itemName.isEmpty()) {
-            return itemName.matches("[a-zA-Z0-9_]*");
-        }
-
-        return false;
+        return StringUtils.isNotEmpty(itemName) && itemName.matches("[a-zA-Z0-9_]*");
     }
 
     /**
@@ -75,14 +76,18 @@ public class ItemUtil {
         }
     }
 
-    public static State convertToAcceptedState(final State state, final Item item) {
+    public static State convertToAcceptedState(State state, Item item) {
+        if (state == null) {
+            LOGGER.error("A conversion of null was requested", new NullPointerException("state should not be null"));
+            return UnDefType.NULL;
+        }
+
         if (item != null && !isAccepted(item, state)) {
             for (Class<? extends State> acceptedType : item.getAcceptedDataTypes()) {
                 State convertedState = state.as(acceptedType);
                 if (convertedState != null) {
-                    LoggerFactory.getLogger(ItemUtil.class).debug("Converting {} '{}' to {} '{}' for item '{}'",
-                            state.getClass().getSimpleName(), state.toString(),
-                            convertedState.getClass().getSimpleName(), convertedState.toString(), item.getName());
+                    LOGGER.debug("Converting {} '{}' to {} '{}' for item '{}'", state.getClass().getSimpleName(), state,
+                            convertedState.getClass().getSimpleName(), convertedState, item.getName());
                     return convertedState;
                 }
             }

